hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkMergeHelper.java

```diff
@@ -77,14 +89,39 @@ public void runMerge(HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>
       readSchema = mergeHandle.getWriterSchemaWithMetaFields();
     }
 
+    Option<InternalSchema> querySchemaOpt = SerDeHelper.fromJson(table.getConfig().getInternalSchema());
+    Boolean needToReWriteRecord = false;
+    // to do support bootstrap
+    if (querySchemaOpt.isPresent() && !baseFile.getBootstrapBaseFile().isPresent()) {
```

Comment:
Should we also check for querySchema not same as dummy schema here ?

Reply:
SerDeHelper.fromJson do not return dummySchema，so it's ok
let me fixed it

Reply:
fixed